src/overlays/actors/ovl_Obj_Dora/z_obj_dora.c

```diff
@@ -28,104 +47,307 @@ const ActorInit Obj_Dora_InitVars = {
+
+void ObjDora_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
+    s32 i;
+    s32 j;
+    Vec3f vtx[3];
+    s32 buf = 0;
+    s32 buff2 = 0;
+
+    Actor_SetScale(&this->actor, 0.1f);
+    ActorShape_Init(&this->actor.shape, 0.0f, &ActorShadow_DrawCircle, 36.0f);
+
+    this->gongRotation.x = 0;
+    this->gongRotation.z = 0;
+    this->gongAngle.x = 0;
+    this->gongAngle.z = 0;
+    this->lastGongHitType = DORA_HIT_NONE;
+    this->rupeeDropTimer = 0;
+    this->collisionCooldownTimer = 0;
+    this->unk3AA = 0;
+    this->gongForce.x = 0.0f;
+    this->gongForce.y = 0.0f;
+    this->gongForce.z = 0.0f;
+
+    Collider_InitTris(globalCtx, &this->colliderTris);
+    Collider_SetTris(globalCtx, &this->colliderTris, &this->actor, &sTrisInit, this->colliderTrisElements);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, &sDamageTable, &sColChkInfoInit);
+
+    Matrix_SetStateRotationAndTranslation(this->actor.world.pos.x, this->actor.world.pos.y, this->actor.world.pos.z,
+                                          &this->actor.shape.rot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderTrisElements); i++) {
+        for (j = 0; j < ARRAY_COUNT(vtx); j++) {
+            Matrix_MultiplyVector3fByState(&sTrisElementsInit[i].dim.vtx[j], &vtx[j]);
+        }
+        Collider_SetTrisVertices(&this->colliderTris, i, &vtx[0], &vtx[1], &vtx[2]);
+    }
+    ObjDora_SetupWait(this);
+}
+
+void ObjDora_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjDora* this = THIS;
+
+    Collider_DestroyTris(globalCtx, &this->colliderTris);
+}
+
+void ObjDora_SetupWait(ObjDora* this) {
+    this->actionFunc = ObjDora_Wait;
+}
+
+void ObjDora_Wait(ObjDora* this, GlobalContext* globalCtx) {
+}
+
+void ObjDora_SetupMoveGong(ObjDora* this) {
+    if (this->lastGongHitType == DORA_HIT_LIGHT) {
+        this->gongForce.x = 2.0f;
+        this->gongForce.y = 5461.0f;
+        this->gongForce.z = 1820.0f;
+    } else {
+        this->gongForce.x = 4.0f;
+        this->gongForce.y = 12743.0f;
+        this->gongForce.z = 5461.0f;
+    }
+    this->gongAngle.x = 0;
+    this->gongAngle.z = 0;
+    this->actionFunc = ObjDora_MoveGong;
+}
+
+void ObjDora_MoveGong(ObjDora* this, GlobalContext* globalCtx) {
+
+    if ((this->gongForce.y < 182.0f) && (this->gongForce.z < 182.0f)) {
+        this->lastGongHitType = DORA_HIT_LIGHT;
+        ObjDora_SetupWait(this);
+    }
+
+    Math_SmoothStepToF(&(this->gongForce.x), 0, 0.2f, 0.2f, 0.1f);
+    Math_SmoothStepToF(&(this->gongForce.y), 0, 0.5f, 54.0f, 18.0f);
+    Math_SmoothStepToF(&(this->gongForce.z), 0, 0.5f, 54.0f, 18.0f);
+    this->gongAngle.x += 0x1555;
+    this->gongAngle.z += 0x238E;
+    this->gongRotation.x = Math_SinS(this->gongAngle.x) * this->gongForce.y;
+    this->gongRotation.z = Math_SinS(this->gongAngle.z) * this->gongForce.z;
+}
+
+s32 ObjDora_IsHalfHourly(u16 time) {
```

Comment:
```suggestion
s32 ObjDora_IsHalfHour(u16 time) {
```
Up to you, but I think this sounds/reads better.